ts/nni_manager/training_service/reusable/gpuScheduler.ts

```diff
@@ -190,10 +191,30 @@ export class GpuScheduler {
             return randomSelect(qualifiedEnvironments);
         } else if (this.policyName === 'round-robin') {
             return this.roundRobinSelect(qualifiedEnvironments, allEnvironments);
+        } else if (this.policyName === 'recently-idle') {
+            return this.recentlyIdleSelect(qualifiedEnvironments, allEnvironments);
         } else {
             throw new Error(`Unsupported schedule policy: ${this.policyName}`);
         }
     }
+    
+    // Select the environment which is idle most recently. If all environments are not idle, use round robin to select an environment.
+    private recentlyIdleSelect(qualifiedEnvironments: EnvironmentInformation[], allEnvironments: EnvironmentInformation[]) : EnvironmentInformation {
+        const now = Date.now();
+        let selectedEnvironment: EnvironmentInformation | undefined = undefined;
+        let minTimeInterval = Number.MAX_SAFE_INTEGER;
+        for (let environment of qualifiedEnvironments) {
+            if (environment.latestTrialReleasedTime > 0 && (now - environment.latestTrialReleasedTime) < minTimeInterval) {
+                selectedEnvironment = environment;
+                minTimeInterval = now - environment.latestTrialReleasedTime;
+            }
+        }
+        if (selectedEnvironment === undefined) {
+            return this.roundRobinSelect(qualifiedEnvironments, allEnvironments);
+        }
+        selectedEnvironment.latestTrialReleasedTime = -1;
```

Comment:
BTW, one environment only runs one trial at a time?

Reply:
if it is, then policy looks good. if not, one environment can run two trials concurrently, then this policy becomes round robin

Reply:
It could run multiple trial at one environment, why it will become round-robin?